nilearn/decomposition/dict_learning.py

```diff
@@ -0,0 +1,171 @@
+                 n_jobs=1, verbose=0,
+                 ):
+        CanICA.__init__(self,
+                        mask=mask, memory=memory, memory_level=memory_level,
+                        n_jobs=n_jobs, verbose=verbose, do_cca=True,
+                        threshold=float(n_components), n_init=1,
+                        n_components=n_components, smoothing_fwhm=smoothing_fwhm,
+                        target_affine=target_affine, target_shape=target_shape,
+                        random_state=random_state, high_pass=high_pass, low_pass=low_pass,
+                        t_r=t_r,
+                        standardize=standardize)
+        self._keep_data_mem = True
+        # Setting n_jobs = 1 as it is slower otherwise
+        MiniBatchDictionaryLearning.__init__(self, n_components=n_components, alpha=alpha,
+                                             n_iter=n_iter, batch_size=10,
+                                             fit_algorithm='lars',
+                                             transform_algorithm='lasso_lars',
+                                             transform_alpha=alpha,
+                                             verbose=verbose,
+                                             random_state=random_state,
+                                             shuffle=True,
+                                             n_jobs=1)
+
+    def _init_dict(self, imgs, y=None, confounds=None):
+        CanICA.fit(self, imgs, y=y, confounds=confounds)
+        if isinstance(self.data_flat_, tuple):  # several subjects
+            self.data_flat_ = np.concatenate(self.data_flat_, axis=0)
+        if self.n_iter == 'auto':
+            # ceil(self.data_fat.shape[0] / self.batch_size)
+            self.n_iter = (self.data_flat_.shape[0] - 1) / self.batch_size + 1
+        if self.verbose:
+            print('[DictLearning] Learning time serie')
+        ridge = Ridge(alpha=1e-6, fit_intercept=None)
+        ridge.fit(self.components_.T, self.data_flat_.T)
+        self.dict_init = ridge.coef_.T
+        S = np.sqrt(np.sum(self.dict_init ** 2, axis=0))
+        self.dict_init /= S[np.newaxis, :]
+
+    def fit(self, imgs, y=None, confounds=None):
+        """Compute the mask and the ICA maps across subjects
+
+        Parameters
+        ----------
+        imgs: list of Niimg-like objects
+            See http://nilearn.github.io/building_blocks/manipulating_mr_images.html#niimg.
+            Data on which PCA must be calculated. If this is a list,
+            the affine is considered the same for all.
+
+        confounds: CSV file path or 2D matrix
+            This parameter is passed to nilearn.signal.clean. Please see the
+            related documentation for details
+        """
+        self._init_dict(imgs, y, confounds)
+
+        if self.verbose:
+            print('[DictLearning] Learning dictionary')
+        MiniBatchDictionaryLearning.fit(self, self.data_flat_.T)
```

Comment:
I would think that here you can use the function 'online_dict_learning' from scikit-learn, rather than the object. Is there a reason that you are using the object?